stage/configuration/test_directory_origin.py

```diff
@@ -397,10 +397,48 @@ def test_directory_origin_configuration_delimited_messages(sdc_builder, sdc_exec
 
 @pytest.mark.parametrize('delimiter_format_type', ['CUSTOM'])
 @pytest.mark.parametrize('data_format', ['DELIMITED'])
-@pytest.mark.skip('Not yet implemented')
+@pytest.mark.parametrize('delimiter_character', [' ', '^'])
+@pytest.mark.parametrize('root_field_type', ['LIST_MAP'])
+@pytest.mark.parametrize('header_line', ['WITH_HEADER'])
 def test_directory_origin_configuration_delimiter_character(sdc_builder, sdc_executor,
-                                                            delimiter_format_type, data_format):
-    pass
+                                                            delimiter_format_type, data_format,
+                                                            delimiter_character, shell_executor, delimited_file_writer,
+                                                            root_field_type, header_line):
+    """Test for Directory origin can read delimited file with custom delimiter character format type.
+        Here we will be creating delimited files with different delimiter character for testing. e.g. [' ', '^']
+    """
+    files_directory = os.path.join('/tmp', get_random_string())
+    FILE_NAME = 'custom_delimited_file.csv'
+    FILE_CONTENTS = [['field1', 'field2', 'field3'], ['Field11', 'Field12', 'fält13'], ['стол', 'Field22', 'Field23']]
```

Comment:
It took me sometime to understand what is happening in the test. What I think would be easy to understand is may be call the first list elements as headers like:
```suggestion
    FILE_CONTENTS = [['header1', 'header2', 'header3'], ['Field11', 'Field12', 'fält13'], ['стол', 'Field22', 'Field23']]
```